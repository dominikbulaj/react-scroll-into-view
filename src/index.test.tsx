import React from 'react'
import ScrollInto from './index'
import renderer from 'react-test-renderer'
import { render, fireEvent, screen } from '@testing-library/react'

test('ScrollInto renders with children', () => {
  const component = renderer.create(
    <ScrollInto selector="test">
      <div>Some child content</div>
    </ScrollInto>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('ScrollInto renders with custom className', () => {
  const component = renderer.create(
    <ScrollInto selector="test" className="custom-class">
      <div>Some child content</div>
    </ScrollInto>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('ScrollInto renders with custom styles', () => {
  const component = renderer.create(
    <ScrollInto selector="test" style={{ display: 'inline' }}>
      <div>Some child content</div>
    </ScrollInto>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('ScrollInto renders with custom styles and className', () => {
  const component = renderer.create(
    <ScrollInto
      selector="test"
      style={{ display: 'inline', color: 'red' }}
      className="pinky"
    >
      <div>Some child content</div>
    </ScrollInto>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Uses provided callback function', () => {
  const mockFn = jest.fn()
  const wrapper = render(
    <ScrollInto selector="test" onClick={mockFn}>
      abc
    </ScrollInto>
  )
  expect(mockFn).toBeCalledTimes(0)

  fireEvent.click(screen.getByText('abc'))
  // wrapper.find('ScrollInto').simulate('click')
  expect(mockFn).toBeCalled()
  expect(mockFn).toBeCalledTimes(1)
})
