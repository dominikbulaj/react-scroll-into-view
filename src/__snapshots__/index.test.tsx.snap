// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollInto renders with children 1`] = `
<div
  className=""
  onClick={[Function]}
  style={Object {}}
>
  <div>
    Some child content
  </div>
</div>
`;

exports[`ScrollInto renders with custom className 1`] = `
<div
  className="custom-class"
  onClick={[Function]}
  style={Object {}}
>
  <div>
    Some child content
  </div>
</div>
`;

exports[`ScrollInto renders with custom styles 1`] = `
<div
  className=""
  onClick={[Function]}
  style={
    Object {
      "display": "inline",
    }
  }
>
  <div>
    Some child content
  </div>
</div>
`;

exports[`ScrollInto renders with custom styles and className 1`] = `
<div
  className="pinky"
  onClick={[Function]}
  style={
    Object {
      "color": "red",
      "display": "inline",
    }
  }
>
  <div>
    Some child content
  </div>
</div>
`;
